### ISSUE TITLE
[Bug] missing documentation on FastCounterInterface constraints

### ISSUE THREAD

**Issue by qku** [Ulm-IQO/qudi-iqo-modules#159]
### Version

current main

### What is affected by the bug?

The constraints for fast counter hardware are a dictionary. The keys are nowhere specified however. Therefore I am having difficulty in writing a hardware module implementing the `FastCounterInterface`.

### When does the bug occur?

not applicable

### How do we replicate the issue?

not applicable

### Expected behavior

The structure of the dictionary returned by `get_constraints()` is documented in the interface, e.g. in the docstring of this method.

### Additional Comments

Looking through existing hardware implementations, I could find the following keys in `FastComtec`:

```python
constraints['hardware_binwidth_list'] = list(self.minimal_binwidth * (2 ** np.array(np.linspace(0,24,25))))
constraints['max_sweep_len'] = 6.8
constraints['max_bins'] = 6.8 / 0.2e-9
```

If this implementation is correct, the docstring for get_constraints() is actually incorrect, since it states:

> The items of the keys are again dictionaries which have the generic
>         dictionary form:
>             {'min': <value>,
>              'max': <value>,
>              'step': <value>,
>              'unit': '<value>'}
> 
> Only the key 'hardware_binwidth_list' differs, since they
> contain the list of possible binwidths.

**Comment by timoML** [comment]
I started overhauling the FastCounterInterface in [this branch](https://github.com/Ulm-IQO/qudi-iqo-modules/tree/fastcounter_constrains).

**Comment by qku** [comment]
@timoML I have two questions on the fast counter interface:

1. How should `stop_measure` and `pause_measure` behave differently? Should the data on the device be cleared if `stop_measure` was called?
2. Should the data also be cleared on the device after `get_data_trace` was called?

**Comment by Neverhorst** [comment]
> For the time tagger we use in the lab (quTAG from qutools), `hardware_binwidth_list` does not make a lot of sense actually. The device can do any multiple of 1 ps as a binwidth.
> 

That depends on the hardware implementation. If the hardware directly samples the signal with a clock, you usually end up with an integer fractional (M/D) of a base clock frequency (non-linear increments).
This is the standard method in older devices.
The newer ones usually sample with maximum accuracy and implement software binning, which enables linear increments.

A revision of this constraint should introduce case sensitive handling to cover both cases.
The easier and not necessarily worse solution would be to keep it a discrete set of allowed values since this works for both cases.

**Comment by timoML** [comment]
I agree, the reworked FastCounterInterface should definitely have better support for linear binwidth increments.
I see two solutions:
1. Linear increment could be a constraints flag that changes the [behavior of the pulsed_gui](https://github.com/Ulm-IQO/qudi-iqo-modules/blob/3c6b3bc42e16e79e0d1181ccc0ef0af12cc0ba2a/src/qudi/gui/pulsed/pulsed_maingui.py#L2432), and replace the binwidth comboBox with a single float input.
2. Hardware with linear increment could emulate the current behavior by a "lazy" list object that makes use of a generator. This "list" could be very long or even infinite and the pulsed_gui could show the first ~100 binwidth list elements. This would come with the drawback that free configuration of the binwidth (with values outside of the binwidth list elements sampled in the combobox) is only possible by a script that call the [pulsed_logic.set_binwidth()](https://github.com/Ulm-IQO/qudi-iqo-modules/blob/3c6b3bc42e16e79e0d1181ccc0ef0af12cc0ba2a/src/qudi/logic/pulsed/pulsed_measurement_logic.py#L358).
